### DISCUSSION TITLE
LocalAPI with React Native

### DISCUSSION THREAD

**Discussion by tommyboylab** [payloadcms/payload#5020] [Q&A]
I'm wondering if anyone has an example of using the LocalAPI within an React Native (Expo) application. I currently have a monorepo with Payload and Expo and was hoping to utilize the LocalAPI for data fetching.

**Comment by ccssmnn** [comment]
The LocalAPI can be used in the same server environment where payload itself is served. There and only there you can skip the HTTP-overhead since you're on the server already.

Your React Native App must use the REST or GraphQL endpoints.

**Reply by kiriti999** [reply]
@tommyboylab But expo itself is fullstack framework for mobile apps so how did you utilize payload then?

**Comment by davelsio** [comment]
Hey there! It's nice to see more folks building with React Native + Payload.

If you are using [Expo Router 3](https://expo.dev/changelog/2024/01-23-router-3), you could theoretically use the new API routes, similarly to what Next.js does.

Though, as already mentioned, you'd have to host Payload and the Expo server runtime in the same place. You can find more info about this on the [API routes docs](https://docs.expo.dev/router/reference/api-routes/#deployment).
